src/main/java/com/google/netpcapanalysis/dao/MaliciousIPDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.google.netpcapanalysis.dao;
+
+import com.google.netpcapanalysis.interfaces.dao.MaliciousIPDao;
+import com.google.netpcapanalysis.models.PCAPdata;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import java.io.IOException;
+
+import javax.lang.model.util.ElementScanner6;
+
+public class  MaliciousIPDaoImpl implements MaliciousIPDao{
+    private static final String AUTH0_API_KEY = "replace_api_key";
+
+    private static final String FLAGGED_FALSE = "Resource not found";
+    private static final String FLAGGED_TRUE = "200: OK";
+    private static final String REQUEST_LIMIT = "Rate limit exceeded.Please reduce your hits per minute.";
+
+    public MaliciousIPDaoImpl(){
+
+    }
+    public PCAPdata isMalicious(PCAPdata data)
+    {
+        HttpResponse<String> result;
+        try {
+            result = Unirest.get("https://signals.api.auth0.com/badip/" + data.destination)
+                    .header("X-Auth-Token", AUTH0_API_KEY)
+                    .asString();
+       
+            if(result.getBody().equals(FLAGGED_FALSE))
+            {
+                data.flagged = false;
+            }
+            else if(result.getBody().equals(FLAGGED_TRUE))
+            {
+                data.flagged = true;
+            }
+            else if(result.getBody().equals(REQUEST_LIMIT))
+            {//documentations doesnt list what per minute limit is
+                data.flagged = false;
```

Comment:
This is good. But you might want to tag it as UNKNOWN . So instead true false use a ENUM.

Reply:
Data store does not support ENUM, only primitive datatypes int, string, bool. Changed flagged to string and added static values

Reply:
yes, typically datastore wont support enums, you store them as int and then the code you map them to enum for readability. But static string is fine too.